src/backend/joanie/core/utils/payment_schedule.py

```diff
@@ -51,7 +51,9 @@ def _withdrawal_limit_date(signed_contract_date):
     withdrawal_date = signed_contract_date + timedelta(days=16)
     if not calendar.is_working_day(withdrawal_date):
         return calendar.add_working_days(withdrawal_date, 1, keep_datetime=True)
-    return withdrawal_date
+    return (
+        withdrawal_date if withdrawal_date < course_start_date else signed_contract_date
+    )
```

Comment:
Should we not do that only if the user explicitly asked to revoke his right of withdrawal ?

Reply:
In this scenario he doesn't have the choice. We must add in the workflow this step where the user explicitly accept it or he can not continue.